esx_service/cli/vmdkops_admin.py

```diff
@@ -723,38 +731,341 @@ def get_version():
     except:
         return NOT_AVAILABLE
 
+def get_tenant_from_db(name):
+    auth.connect_auth_db()
+    error_info, tenant = auth._auth_mgr.get_tenant(name)
+    return error_info, tenant
+
+def operation_fail(error_info):
+    print error_info
+    return error_info
+
+def tenant_ls_headers():
+    """ Return column names for tenant ls command """
+    headers = ['Uuid', 'Name', 'Description', 'Default_datastore', 'VM_list']
+    return headers
+
+def generate_vm_list(vms):
+    """ Generate a comma separated string with given list """
+    res = ""
+    for vm in vms:
+        # vm[1] is vm_name, vm has format (vm_uuid, vm_name)
+        res = res + vm[1]
+        res = res + ","
+        
+    if res:
+        res = res[:-1]
+
+    return res    
+
+def generate_tenant_ls_rows(tenant_list):
+    """ Generate output for tenant ls command """
+    rows = []
+    for tenant in tenant_list:
+        uuid = tenant.id
+        name = tenant.name
+        description = tenant.description
+        default_datastore = tenant.default_datastore
+        vm_list = generate_vm_list(tenant.vms)
+        rows.append([uuid, name, description, default_datastore, vm_list])
+    
+    return rows
 
-NOT_IMPLEMENTED = "Not implemented"
+def generate_tuple_from_vm_list(vm_list):
+    """ Generate a list of (vm_uuid, vm_name) pair """
+    vms = []
+    for vm_name in vm_list:
+        vm_uuid = vmdk_utils.get_vm_uuid_by_name(vm_name)
+        if not vm_uuid:
+            error_info =  "Cannot find vm_uuid for vm {0}, tenant_create failed".format(vm_name)
+            return error_info, None
+        vms.append((vm_uuid, vm_name))
+    
+    return None, vms
 
 def tenant_create(args):
-    print(NOT_IMPLEMENTED)
-
+    """ Handle tenant create command """
+    error_info, vms = generate_tuple_from_vm_list(args.vm_list)
+    if error_info:
+        return operation_fail(error_info)
+      
+    name = args.name 
+    description = "" 
+    default_datastore ="default_ds" 
+    default_privileges =  {}
+    privileges = []
+
+    auth.connect_auth_db()
```

Comment:
it would be good to have all connect_auth_db() wrapped the same way get_tenant_from_db() is wrapping it - this will place all connection mgmt to the same software layer.